packages/xstate-react/src/fsm.ts

```diff
@@ -56,58 +56,54 @@ export function useMachine<
         stateMachine.config,
         options ? options : (stateMachine as any)._options
       )
-    ).start()
+    )
   );
 
   const [state, setState] = useState(() => getServiceState(service));
 
-  useEffect(() => {
+  useIsomorphicLayoutEffect(() => {
     if (options) {
       (service as any)._machine._options = options;
     }
   });
 
   useEffect(() => {
     service.subscribe(setState);
+    service.start();
```

Comment:
This aligns the behavior with the same logic that we have for the XState itself. I will also add a test covering this